test/property/local_watcher/index.js

```diff
@@ -0,0 +1,127 @@
+const { describe, it } = require('mocha')
+const should = require('should')
+
+const crypto = require('crypto')
+const fse = require('fs-extra')
+const glob = require('glob')
+const path = require('path')
+
+const { ContextDir } = require('../../support/helpers/context_dir')
+const TmpDir = require('../../support/helpers/TmpDir')
+
+const { id } = require('../../../core/metadata')
+const { defaultLogger } = require('../../../core/logger')
+const Ignore = require('../../../core/ignore')
+const Merge = require('../../../core/merge')
+const Pouch = require('../../../core/pouch')
+const Prep = require('../../../core/prep')
+const Watcher = require('../../../core/local/watcher')
+
+const PouchDB = require('pouchdb')
+PouchDB.plugin(require('pouchdb-adapter-memory'))
+
+const events = {
+  emit: msg => {}
+}
+
+async function play (state, op) {
+  // Slow down things to avoid issues with chokidar throttler
+  await Promise.delay(10)
+
+  switch (op.op) {
+    case 'start':
+      const config = { dbPath: { name: state.name, adapter: 'memory' } }
+      state.pouchdb = new Pouch(config)
+      await state.pouchdb.addAllViewsAsync()
+      const merge = new Merge(state.pouchdb)
+      const ignore = new Ignore([])
+      const prep = new Prep(merge, ignore, config)
+      state.watcher = new Watcher(state.dir.root, prep, state.pouchdb, events)
+      state.watcher.start()
+      break
+    case 'sleep':
+      await Promise.delay(op.duration)
+      break
+    case 'mkdir':
+      try {
+        await state.dir.ensureDir(op.path)
+      } catch (err) {}
```

Comment:
> Does errors happen sometimes?

Yes, the generated sequence of operations are not always valid. It's a lot easier to ignore errors when playing the operations that trying to have a sequence that can't fail but still plays with special cases.

> Shouldn't we log them?

No, they are not interesting and they have no effect on the test.

> Could bluebird's #catchReturn() be of some help?

It can help to make the code shorter.